From 04834cffc5eca7119f52e8719e46f263ec0d84a2 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Thu, 24 May 2012 19:12:02 +0800
Subject: ENGR00182098 MX6DL 480P packet overflow interrupt can't been clean

Fix it with not SW reset TMDS when clean overflow interrupt on Mx6DL

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxc_hdmi.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/mxc_hdmi.c b/drivers/video/mxc_hdmi.c
index f486d9a..aea9ffe 100644
--- a/drivers/video/mxc_hdmi.c
+++ b/drivers/video/mxc_hdmi.c
@@ -58,6 +58,7 @@
 
 #include <linux/mfd/mxc-hdmi-core.h>
 #include <mach/mxc_hdmi.h>
+#include <mach/hardware.h>
 
 #define DISPDRV_HDMI	"hdmi"
 #define HDMI_EDID_LEN		512
@@ -1490,7 +1491,8 @@ static void mxc_hdmi_clear_overflow(void)
 		hdmi_writeb(val, HDMI_FC_INVIDCONF);
 
 	/* TMDS software reset */
-	hdmi_writeb((u8)~HDMI_MC_SWRSTZ_TMDSSWRST_REQ, HDMI_MC_SWRSTZ);
+	if (!cpu_is_mx6dl())
+		hdmi_writeb((u8)~HDMI_MC_SWRSTZ_TMDSSWRST_REQ, HDMI_MC_SWRSTZ);
 }
 
 static void hdmi_enable_overflow_interrupts(void)
@@ -1505,6 +1507,7 @@ static void hdmi_disable_overflow_interrupts(void)
 	pr_debug("%s\n", __func__);
 	hdmi_writeb(HDMI_IH_MUTE_FC_STAT2_OVERFLOW_MASK,
 		    HDMI_IH_MUTE_FC_STAT2);
+	hdmi_writeb(0xff, HDMI_FC_MASK2);
 }
 
 static void mxc_hdmi_notify_fb(struct mxc_hdmi *hdmi)
-- 
1.7.9.5

